src/lib/languageUtils.ts

```diff
@@ -1,2 +1,29 @@
 export const CONFIG_NAME_REGEX = /[a-zA-Z0-9_ ]+/;
 export const CONFIG_NAME_WITH_QUOTES_REGEX = /["'`][a-zA-Z0-9_ ]+["'`]/;
+
+const SUPPORTED_LANGUAGES = ['javascript', 'typescript'] as const;
+export type SupportedLanguageType = typeof SUPPORTED_LANGUAGES[number];
+
+export function isLanguageSupported(language: string): boolean {
+  return (SUPPORTED_LANGUAGES as unknown as string[]).includes(language);
+}
+
+export function getStatsigAPICheckGateRegex(
+  language: SupportedLanguageType,
+): RegExp {
+  switch (language) {
+    case 'javascript':
+    case 'typescript':
+      return /[a-zA-Z0-9_]+[.]+checkGate+\(+[^)]+\)/;
+  }
+}
+
+export function getStatsigAPIGetConfigRegex(
+  language: SupportedLanguageType,
+): RegExp {
+  switch (language) {
+    case 'javascript':
+    case 'typescript':
+      return /[a-zA-Z0-9_]+[.]+getConfig+\(+[^)]+\)/;
```

Comment:
`getConfig` or `getExperiment`, both return a `DynamicConfig`

Reply:
Oh interesting, so the "dynamic_configs" from DCS can be referenced by both `getConfig` and `getExperiment`? Or do you mean they share the same base `Ent`

Reply:
Well, both.  getExperiment and getConfig do mostly the same things (both return a DynamicConfig and they are mostly interchangeable, there are some special parameters for getExperiment in client SDKs (sticky experiments).

Can you add getExperiment? Still didnt see that